src/internal/operators/shareReplay.ts

```diff
@@ -126,18 +126,20 @@ export function shareReplay<T>(
 ): MonoTypeOperatorFunction<T> {
   let bufferSize: number;
   let refCount = false;
-  if (configOrBufferSize && typeof configOrBufferSize === 'object') {
+  if (!configOrBufferSize) {
+    bufferSize = Infinity;
```

Comment:
AFAICT, this will effect different behaviour, as passing a buffer size of zero will result in an infinite buffer.